### ISSUE TITLE
Save and Load x.mat for QC output after each ASL submodule

### ISSUE THREAD

**Issue by jan-petr** [ExploreASL/ExploreASL#121]
* Load and save the x.mat to keep track of Output before and after each wrapper.

* Make sure that no spurious x is added in either of the submodules that would not be worthy of saving (in most submodules, the x modifications were discarded and not returned, so there can be some weird thins - although in some modules, this is returned).

* Move some of the QC things to the correct submodule.

**Comment by HenkMutsaerts** [comment]
You fixed this right, should we move this issue to milestone 1.2.1 and close it?

**Comment by jan-petr** [comment]
No. I have fixed it for one ASL submodule only, but not for all the other submodules.

As written in the description - the other submodules only modify x inside, but throw away all the changes. So I have to write loading and saving of x before/after each submodule. But at the same time, I have to verify that nothing important is changed inside the x. Right now, the x-struct changes are always discarded, so we can't just output and save it without checking.

So:
1) This needs some time, but is not at all urgent.
2) After done, needs proper testing to avoid introducing bugs - i.e. not good to do it in a hotfix.

**Comment by jan-petr** [comment]
### Reworked ToDo after discussion in #112 and #122:

* Load and save the QC*.json to keep track of Output before and after each submodule wrapper in the ASL module. Replaces the loading and saving of x.mat.
* x.mat is currently saved for xASL_wrp_registerASL, but this can be removed and replaced by QC*.json, as the x.mat was saved solely for the purpose of QC and not for other things in the x.mat.
* At the start of ASL module - next to the output cleaning in x.mat, make sure that the QC*json is cleaned and created as well.
* Move some of the QC things to the correct submodule.
* The question still is, if the same output is saved to x.mat in parallel, or kept only in QC*.json?

**Comment by HenkMutsaerts** [comment]
* OK, then we only keep QC*.json
* we open and close QC*.json each time, and don't keep track what was opened and closed. Each submodule should be "on its own" in this regard.
* In the derivative stuff that now loads the x.mat (e.g. PDF), we should convert to QC*.json as well.
* Can become confusing that part of the output is moved to json, but the images are still inside the x.mat 
(there are both x.Output.(ScanType).* and x.Output_im.(ScanType).*

**Comment by HenkMutsaerts** [comment]
## Copy of my comment before closing #112
>I agree to write the QC*.json instead of the x.mat. That will make stuff much easier, as the changes in the x.mat are almost always thrown away after each submodule.

Correction: the changes in x.mat are not thrown away, but the changes in x, right? At least that's the idea. Anyway, I agree changing QC*.json is more specific.
>There's only one question then - should this QC*.json loading and saving happen inside the submodule. Or should the submodule output a QC structure, which is then added to the QC*.json. I support the second option and I wrote it like that to the #121 - can you please verify that there while we have it fresh in our heads so that we reach an agreement that I can later implement?

I agree with you here. This should be standardized across all submodules.
Will merge this now!
